src/test/scala/org/hammerlab/guacamole/reads/ReadSetSuite.scala

```diff
@@ -20,36 +20,75 @@ package org.hammerlab.guacamole.reads
 
 import org.apache.parquet.hadoop.metadata.CompressionCodecName
 import org.bdgenomics.adam.rdd.read.AlignmentRecordRDDFunctions
-import org.bdgenomics.adam.rdd.{ ADAMSaveAnyArgs, ADAMContext }
-import org.hammerlab.guacamole.reference.ReferenceBroadcast
-import org.hammerlab.guacamole.{ LociSet }
+import org.bdgenomics.adam.rdd.{ADAMContext, ADAMSaveAnyArgs}
+import org.hammerlab.guacamole.LociSet
 import org.hammerlab.guacamole.reads.Read.InputFilters
-import org.hammerlab.guacamole.util.{ TestUtil, GuacFunSuite }
+import org.hammerlab.guacamole.reference.ReferenceBroadcast
+import org.hammerlab.guacamole.util.{GuacFunSuite, TestUtil}
 import org.scalatest.Matchers
 
 class ReadSetSuite extends GuacFunSuite with Matchers {
 
   def chr22Fasta = ReferenceBroadcast.readFasta(TestUtil.testDataPath("chr22.fa.gz"), sc)
 
+  case class LazyMessage(msg: () ⇒ String) {
+    override def toString: String = msg()
+  }
+
   sparkTest("using different bam reading APIs on sam/bam files should give identical results") {
     def check(paths: Seq[String], filter: InputFilters): Unit = {
       withClue("using filter %s: ".format(filter)) {
-        val configs = Read.ReadLoadingConfig.BamReaderAPI.values.map(api => Read.ReadLoadingConfig(bamReaderAPI = api)).toSeq
-        val standard = TestUtil.loadReads(sc, paths(0), filter, config = configs(0), reference = chr22Fasta).reads.collect
-        configs.foreach(config => {
-          paths.foreach(path => {
-            withClue("file %s with config %s vs standard %s with config %s:\n".format(path, config, paths(0), configs(0))) {
-              val result = TestUtil.loadReads(sc, path, filter, config = config, reference = chr22Fasta).reads.collect
-              if (result.toSet != standard.toSet) {
-                val missing = standard.filter(!result.contains(_))
-                assert(missing.isEmpty, "Missing reads: %s".format(missing.map(_.toString).mkString("\n\t")))
-                val extra = result.filter(!standard.contains(_))
-                assert(extra.isEmpty, "Extra reads:\n\t%s".format(extra.map(_.toString).mkString("\n\t")))
-                assert(false, "shouldn't get here")
-              }
-            }
-          })
-        })
+
+        val firstPath = paths.head
+
+        val configs =
+          Read.ReadLoadingConfig.BamReaderAPI.values
+            .map(api => Read.ReadLoadingConfig(bamReaderAPI = api))
+
+        val firstConfig = configs.head
+
+        val standard =
+          TestUtil.loadReads(
+            sc,
+            paths.head,
+            filter,
+            config = configs.head,
+            reference = chr22Fasta
+          ).reads.collect
+
+        for {
+          config ← configs
```

Comment:
These symbols seem like a complexity cost without much benefit to me. They
will look odd to someone new to scala (even someone not so new, like me),
render weirdly on github (the => character looks like a -> to me on my
screen), may be a hassle for non-intellij manipulation of the codebase like
grep, and will likely result in inconsistency where in some parts of the
code we use the symbols and in other parts of the code we do not. They also
evoke the horrors of APL. There are lots of other non-single-character
syntax tokens like 'val', so I don't see a problem with having '->' spelled
out as two characters.

On Fri, Apr 1, 2016 at 2:10 PM, Ryan Williams <notifications@github.com>
wrote:

> In src/test/scala/org/hammerlab/guacamole/reads/ReadSetSuite.scala
> <https://github.com/hammerlab/guacamole/pull/416#discussion_r58244716>:
>
> > +          Read.ReadLoadingConfig.BamReaderAPI.values
> > +            .map(api => Read.ReadLoadingConfig(bamReaderAPI = api))
> > +
> > +        val firstConfig = configs.head
> > +
> > +        val standard =
> > +          TestUtil.loadReads(
> > +            sc,
> > +            paths.head,
> > +            filter,
> > +            config = configs.head,
> > +            reference = chr22Fasta
> > +          ).reads.collect
> > +
> > +        for {
> > +          config ← configs
>
> I think you should reconsider. scalac/zinc, Intellij, (even scalariform!
> <https://github.com/scala-ide/scalariform#rewritearrowsymbols>), all
> support this, and I think it's a clear improvement.
>
> You never mean "dash, greater than", "less than, dash", or "equals,
> greater than", and scala doesn't understand them that way; you mean "right
> arrow", "left arrow", and "double right arrow".
>
> I thought it was a weird idea but after using it the old way seems broken.
> No need to ASCII-art our code where the entire toolchain and language
> idioms have freed us from having to. You don't have to do anything
> different, either, intellij rewrites it for you.
>
> You can also skip the setting, but I'd like to keep it on. lmk what you
> think.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/hammerlab/guacamole/pull/416/files/d497ee828be0253b9289f897dd9016811d90040d#r58244716>
>